crest/Assets/Crest/Crest/Scripts/Helpers/FloatingOrigin.cs

```diff
@@ -65,35 +69,149 @@ public class FloatingOrigin : MonoBehaviour
         [Tooltip("Optionally provide a list of Gerstner components to avoid doing a FindObjectsOfType() call."), SerializeField]
         ShapeGerstnerBatched[] _overrideGerstnerList = null;
 
+        [Space(10)]
+
+        [SerializeField]
+        internal DebugFields _debug = new DebugFields();
+
+        [System.Serializable]
+        internal partial class DebugFields
+        {
+            [Tooltip("GPU capture using external tool on origin shift.")]
+            public bool _captureOnShift;
+            [Tooltip("Pause editor on origin shift.")]
+            public bool _pauseOnShift;
+            [Tooltip("Pause editor before origin shift. When it meets the threshold it postpones the shift to the next frame and pauses this frame.")]
+            public bool _pauseBeforeShift;
+            [Tooltip("Log to console on origin shift.")]
+            public bool _logOnShift;
+            internal bool _isCapturing;
+            internal bool _shiftNextX;
+            internal bool _shiftNextY;
+        }
+
+        bool _isFirstFixedUpdate = true;
+
         ParticleSystem.Particle[] _particleBuffer = null;
 
-        static readonly int sp_CrestFloatingOriginOffset = Shader.PropertyToID("_CrestFloatingOriginOffset");
+        public static readonly int sp_CrestFloatingOriginOffset = Shader.PropertyToID("_CrestFloatingOriginOffset");
 
         Vector3 _originOffset;
 
         public static Vector3 TeleportOriginThisFrame { get; private set; }
 
-        void LateUpdate()
+        IEnumerator Start()
         {
-            TeleportOriginThisFrame = Vector3.zero;
+            while (true)
+            {
+                // NOTE: Will not work in batch mode:
+                // https://docs.unity3d.com/Manual/CLIBatchmodeCoroutines.html
+                yield return Helpers.WaitForEndOfFrame;
+
+                if (_debug._isCapturing)
+                {
+                    ExternalGPUProfiler.EndGPUCapture();
```

Comment:
interesting, nice

Reply:
Indeed. It was very useful/necessary as LOD doesn't run when paused.